lib/errors.js

```diff
@@ -37,17 +37,42 @@ Errors.prototype = {
             line = line.line;
         }
 
-        // line and column numbers should be explicit
-        assert(typeof line === 'number' && line > 0,
-            'Unable to add an error, `line` should be a number greater than 0 but ' + line + ' given');
-        assert(typeof column === 'number' && column >= 0,
-            'Unable to add an error, `column` should be a positive number but ' + column + ' given');
-
-        this._addError({
+        var errorInfo = {
```

Comment:
Yeah, this place is not the best to talk about hidden classes and performance. `errors.cast` can do the job.

Personally I'm trying to keep it in mind. Ok, nvm!